rust/src/hazmat.rs

```diff
@@ -25,7 +29,39 @@ pub mod sha3;
 /// Unified error type.
 #[derive(Debug, PartialEq, Eq, Clone, Copy)]
 pub enum Error {
-    Chacha20Poly1305(chacha20_poly1305::Error),
+    ChaCha20Poly1305(chacha20_poly1305::Error),
     Curve25519(curve25519::Error),
     P256(p256::Error),
+    Ed25519(ed25519::Error),
+    Hkdf(hkdf::Error),
+}
+
+impl Into<Error> for chacha20_poly1305::Error {
```

Comment:
Suggestion: This seems like a good fit for `thiserror`.

Reply:
Yeah, but it's not really worth the dependency.